syft/source/stereoscope/provider.go

```diff
@@ -0,0 +1,76 @@
+package stereoscope
+
+import (
+	"context"
+
+	"github.com/anchore/stereoscope"
+	"github.com/anchore/stereoscope/pkg/image"
+	"github.com/anchore/stereoscope/tagged"
+	"github.com/anchore/syft/syft/source"
+)
+
+const ImageTag = "image"
+
+type stereoscopeSourceProvider struct {
+	stereoscopeProvider image.Provider
+	alias               *source.Alias
+	cfg                 *SourceProviderConfig
+}
+
+var _ source.Provider = (*stereoscopeSourceProvider)(nil)
+
+func (l stereoscopeSourceProvider) Name() string {
+	return l.stereoscopeProvider.Name()
+}
+
+func (l stereoscopeSourceProvider) Provide(ctx context.Context, userInput string) (source.Source, error) {
+	img, err := l.stereoscopeProvider.Provide(ctx, userInput)
+	if err != nil {
+		return nil, err
+	}
+	var alias source.Alias
+	if !l.alias.IsEmpty() {
+		alias = *l.alias
+	}
+	cfg := ImageConfig{
+		Reference:       userInput,
+		From:            l.stereoscopeProvider.Name(),
+		Platform:        l.cfg.Platform,
+		RegistryOptions: l.cfg.RegistryOptions,
+		Exclude:         l.cfg.Exclude,
+		Alias:           alias,
+	}
+	if err != nil {
+		return nil, err
+	}
+	return NewStereoscopeImageSource(img, cfg), nil
+}
+
+type SourceProviderConfig struct {
+	RegistryOptions *image.RegistryOptions
+	Platform        *image.Platform
+	Alias           *source.Alias
+	Exclude         source.ExcludeConfig
+}
+
+func SourceProviders(cfg SourceProviderConfig) tagged.Values[source.Provider] {
```

Comment:
```suggestion
func Providers(cfg SourceProviderConfig) tagged.Values[source.Provider] {
```
given the package rename